src/twisted/conch/interfaces.py

```diff
@@ -82,6 +82,26 @@ def getPty(term, windowSize, modes):
         fails, raise an exception.
         """
 
+    def setEnv(name, value):
+        """
+        Set an environment variable for the shell or command to be started.
+
+        From U{RFC 4254, section 6.4
+        <https://tools.ietf.org/html/rfc4254#section-6.4>}: "Uncontrolled
+        setting of environment variables in a privileged process can be a
+        security hazard.  It is recommended that implementations either
+        maintain a list of allowable variable names or only set environment
+        variables after the server process has dropped sufficient
+        privileges."
+
```

Comment:
It would be useful to mention what the OpenSSH-compatible behavior is here, since any alternative implementation will probably want to implement it.

Reply:
Done.